# 01 — Inventario y validación de fuentes

## Objetivo

Documentar las fuentes de video del MVP de Mitotl IA y validar que la estructura local del proyecto esté preparada para recibir videosde referencia y de ejecución.

## Alcance

- Revisar la estructura de carpetas.
- Identificar videos disponibles.
- Registrar metadatos básicos.
- Documentar fuentes y condiciones de uso.
- Preparar la salida para los siguientes notebooks.

## Limitaciones

AIST se utilizará como fuente técnica de referencia. No representa un conjunto etiquetado para evaluar calidad profesional de danza.

In [12]:
from pathlib import Path
from datetime import datetime

PROJECT_ROOT = Path.cwd().parent
DATA_DIR = PROJECT_ROOT / "data"
REFERENCE_DIR = DATA_DIR / "raw" / "reference"
EXECUTION_DIR = DATA_DIR / "raw" / "execution"
METADATA_DIR = DATA_DIR / "metadata"
ARTIFACTS_DIR = PROJECT_ROOT / "artifacts"

for directory in [
    REFERENCE_DIR,
    EXECUTION_DIR,
    METADATA_DIR,
    ARTIFACTS_DIR,
]:
    directory.mkdir(parents=True, exist_ok=True)

print(f"Proyecto: {PROJECT_ROOT}")
print(f"Referencia: {REFERENCE_DIR}")
print(f"Ejecución: {EXECUTION_DIR}")
print(f"Metadatos: {METADATA_DIR}")
print(f"Artefactos: {ARTIFACTS_DIR}")
print(f"Fecha de revisión: {datetime.now():%Y-%m-%d %H:%M:%S}")

Proyecto: /Users/igayvid/Library/Mobile Documents/com~apple~CloudDocs/Documents/05-aprendizaje/diplomado-ciencia-datos/05-modulo-integrador/mitotl-ia
Referencia: /Users/igayvid/Library/Mobile Documents/com~apple~CloudDocs/Documents/05-aprendizaje/diplomado-ciencia-datos/05-modulo-integrador/mitotl-ia/data/raw/reference
Ejecución: /Users/igayvid/Library/Mobile Documents/com~apple~CloudDocs/Documents/05-aprendizaje/diplomado-ciencia-datos/05-modulo-integrador/mitotl-ia/data/raw/execution
Metadatos: /Users/igayvid/Library/Mobile Documents/com~apple~CloudDocs/Documents/05-aprendizaje/diplomado-ciencia-datos/05-modulo-integrador/mitotl-ia/data/metadata
Artefactos: /Users/igayvid/Library/Mobile Documents/com~apple~CloudDocs/Documents/05-aprendizaje/diplomado-ciencia-datos/05-modulo-integrador/mitotl-ia/artifacts
Fecha de revisión: 2026-08-02 15:29:58


In [13]:
VIDEO_EXTENSIONS = {".mp4", ".mov", ".webm", ".avi", ".mkv"}

reference_files = sorted(
    path for path in REFERENCE_DIR.iterdir()
    if path.is_file() and path.suffix.lower() in VIDEO_EXTENSIONS
)

execution_files = sorted(
    path for path in EXECUTION_DIR.iterdir()
    if path.is_file() and path.suffix.lower() in VIDEO_EXTENSIONS
)

print("Videos de referencia:")
for path in reference_files:
    print(f"- {path.name}")

print("\nVideos de ejecución:")
for path in execution_files:
    print(f"- {path.name}")

if not reference_files:
    print("\nAdvertencia: todavía no hay videos de referencia.")

Videos de referencia:

Videos de ejecución:

Advertencia: todavía no hay videos de referencia.


## Registro de metadatos

Cada video debe registrar su procedencia, permisos y características técnicas. Los archivos de video no se versionarán en Git; únicamente se conservarán localmente y se documentarán mediante metadatos.

In [14]:
from urllib.request import urlopen
import csv
import io

AIST_METADATA_URL = (
    "https://aistdancedb.ongaaccel.jp/"
    "data/video_refined/2M/refined_2M_sBM_all.csv"
)

with urlopen(AIST_METADATA_URL) as response:
    content = response.read().decode("utf-8")

reader = csv.DictReader(io.StringIO(content))
aist_rows = list(reader)

print(f"Registros descargados: {len(aist_rows)}")
print("Columnas disponibles:")
print(reader.fieldnames)

Registros descargados: 10798
Columnas disponibles:
['URL', 'FILE_NAME', 'BASE_NAME', 'SEC', 'FILE_SIZE', 'GENRE', 'SITUATION', 'CAMERA', 'DANCER', 'MUSIC', 'CHOREOGRAPHY']


In [15]:
for row in aist_rows[:3]:
    print(row)

{'URL': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch01.mp4', 'FILE_NAME': 'gBR_sBM_c01_d04_mBR0_ch01.mp4', 'BASE_NAME': 'gBR_sBM_c01_d04_mBR0_ch01', 'SEC': '11.995316666666668', 'FILE_SIZE': '3427560', 'GENRE': 'gBR', 'SITUATION': 'sBM', 'CAMERA': 'c01', 'DANCER': 'd04', 'MUSIC': 'mBR0', 'CHOREOGRAPHY': 'ch01'}
{'URL': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch02.mp4', 'FILE_NAME': 'gBR_sBM_c01_d04_mBR0_ch02.mp4', 'BASE_NAME': 'gBR_sBM_c01_d04_mBR0_ch02', 'SEC': '11.995316666666668', 'FILE_SIZE': '3308958', 'GENRE': 'gBR', 'SITUATION': 'sBM', 'CAMERA': 'c01', 'DANCER': 'd04', 'MUSIC': 'mBR0', 'CHOREOGRAPHY': 'ch02'}
{'URL': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch03.mp4', 'FILE_NAME': 'gBR_sBM_c01_d04_mBR0_ch03.mp4', 'BASE_NAME': 'gBR_sBM_c01_d04_mBR0_ch03', 'SEC': '11.995316666666668', 'FILE_SIZE': '3397453', 'GENRE': 'gBR', 'SITUATION': 'sBM', 'CAMERA': 'c01', 'DANCER': 'd04', 'MUSIC': 'mBR0'

In [16]:
front_camera_rows = [
    row for row in aist_rows
    if "_c01_" in " ".join(row.values())
]

print(f"Videos con cámara c01: {len(front_camera_rows)}")

for row in front_camera_rows[:5]:
    print(row)

Videos con cámara c01: 1200
{'URL': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch01.mp4', 'FILE_NAME': 'gBR_sBM_c01_d04_mBR0_ch01.mp4', 'BASE_NAME': 'gBR_sBM_c01_d04_mBR0_ch01', 'SEC': '11.995316666666668', 'FILE_SIZE': '3427560', 'GENRE': 'gBR', 'SITUATION': 'sBM', 'CAMERA': 'c01', 'DANCER': 'd04', 'MUSIC': 'mBR0', 'CHOREOGRAPHY': 'ch01'}
{'URL': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch02.mp4', 'FILE_NAME': 'gBR_sBM_c01_d04_mBR0_ch02.mp4', 'BASE_NAME': 'gBR_sBM_c01_d04_mBR0_ch02', 'SEC': '11.995316666666668', 'FILE_SIZE': '3308958', 'GENRE': 'gBR', 'SITUATION': 'sBM', 'CAMERA': 'c01', 'DANCER': 'd04', 'MUSIC': 'mBR0', 'CHOREOGRAPHY': 'ch02'}
{'URL': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch03.mp4', 'FILE_NAME': 'gBR_sBM_c01_d04_mBR0_ch03.mp4', 'BASE_NAME': 'gBR_sBM_c01_d04_mBR0_ch03', 'SEC': '11.995316666666668', 'FILE_SIZE': '3397453', 'GENRE': 'gBR', 'SITUATION': 'sBM', 'CAMERA': 'c01', 'DAN

In [17]:
from collections import Counter

print("Situaciones:")
for situation, count in Counter(row["SITUATION"] for row in aist_rows).most_common():
    print(f"{situation}: {count}")

print("\nCámaras:")
for camera, count in Counter(row["CAMERA"] for row in aist_rows).most_common():
    print(f"{camera}: {count}")

print("\nGéneros:")
for genre, count in Counter(row["GENRE"] for row in aist_rows).most_common():
    print(f"{genre}: {count}")

Situaciones:
sBM: 10798

Cámaras:
c01: 1200
c02: 1200
c03: 1200
c04: 1200
c05: 1200
c06: 1200
c07: 1200
c08: 1199
c09: 1199

Géneros:
gBR: 1080
gHO: 1080
gJB: 1080
gJS: 1080
gKR: 1080
gLH: 1080
gLO: 1080
gPO: 1080
gWA: 1080
gMH: 1078


In [18]:
for situation in ["sBM", "sGR"]:
    rows = [
        row for row in aist_rows
        if row["SITUATION"] == situation and row["CAMERA"] == "c01"
    ]

    print(f"{situation} con cámara c01: {len(rows)}")

    for row in rows[:3]:
        print({
            "file_name": row["FILE_NAME"],
            "dancer": row["DANCER"],
            "duration_sec": row["SEC"],
        })

    print()

sBM con cámara c01: 1200
{'file_name': 'gBR_sBM_c01_d04_mBR0_ch01.mp4', 'dancer': 'd04', 'duration_sec': '11.995316666666668'}
{'file_name': 'gBR_sBM_c01_d04_mBR0_ch02.mp4', 'dancer': 'd04', 'duration_sec': '11.995316666666668'}
{'file_name': 'gBR_sBM_c01_d04_mBR0_ch03.mp4', 'dancer': 'd04', 'duration_sec': '11.995316666666668'}

sGR con cámara c01: 0



## Criterios de selección del video de referencia

Para el MVP se seleccionará inicialmente un video que cumpla:

- Una sola persona.
- Cuerpo completo visible.
- Cámara frontal fija (`c01`).
- Video refinado, sin pre-roll ni post-roll.
- Situación `sBM` (Basic Dance).
- Resolución y duración adecuadas para el análisis.

La situación `sGR` (Group Dance) no se utilizará en el MVP porque implica múltiples personas y la estrategia actual delimita el sistema a una sola persona. Su análisis queda reservado para una futura versión.

La selección se basa en el alcance definido del MVP y en los metadatos publicados por AIST, no en una afirmación de que `sBM` sea una categoría superior para evaluar danza.

## Decisión de la fuente de referencia

Se utilizará un video refinado de AIST con las siguientes condiciones:

- Situación: `sBM` (Basic Dance).
- Cámara: `c01` (cámara frontal fija).
- Una persona identificada mediante el campo `DANCER`.
- Video refinado, sin pre-roll ni post-roll.
- Calidad de 2 Mbps para reducir el tamaño de descarga durante el MVP.
- Duración suficiente para probar extracción de pose y comparación.

La selección de `sBM` responde al alcance del MVP, que inicialmente
analiza a una sola persona. Los videos `sGR` se reservan para una futura
versión con detección de múltiples personas.

In [19]:
candidates_by_genre = {}

for row in sorted(front_camera_rows, key=lambda item: item["FILE_NAME"]):
    genre = row["GENRE"]

    if genre not in candidates_by_genre:
        candidates_by_genre[genre] = {
            "file_name": row["FILE_NAME"],
            "url": row["URL"],
            "genre": row["GENRE"],
            "situation": row["SITUATION"],
            "camera": row["CAMERA"],
            "dancer": row["DANCER"],
            "duration_sec": float(row["SEC"]),
            "file_size_bytes": int(row["FILE_SIZE"]),
        }

for candidate in candidates_by_genre.values():
    print(candidate)
    

{'file_name': 'gBR_sBM_c01_d04_mBR0_ch01.mp4', 'url': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gBR_sBM_c01_d04_mBR0_ch01.mp4', 'genre': 'gBR', 'situation': 'sBM', 'camera': 'c01', 'dancer': 'd04', 'duration_sec': 11.995316666666668, 'file_size_bytes': 3427560}
{'file_name': 'gHO_sBM_c01_d19_mHO0_ch01.mp4', 'url': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gHO_sBM_c01_d19_mHO0_ch01.mp4', 'genre': 'gHO', 'situation': 'sBM', 'camera': 'c01', 'dancer': 'd19', 'duration_sec': 8.725392058725392, 'file_size_bytes': 2514732}
{'file_name': 'gJB_sBM_c01_d07_mJB0_ch01.mp4', 'url': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gJB_sBM_c01_d07_mJB0_ch01.mp4', 'genre': 'gJB', 'situation': 'sBM', 'camera': 'c01', 'dancer': 'd07', 'duration_sec': 11.995316666666668, 'file_size_bytes': 3560338}
{'file_name': 'gJS_sBM_c01_d01_mJS0_ch01.mp4', 'url': 'https://aistdancedb.ongaaccel.jp/v1.0.0/video/2M/gJS_sBM_c01_d01_mJS0_ch01.mp4', 'genre': 'gJS', 'situation': 'sBM', 'camera': 'c01', 'd

In [20]:
total_bytes = sum(
    candidate["file_size_bytes"]
    for candidate in candidates_by_genre.values()
)

total_mb = total_bytes / (1024 ** 2)

print(f"Videos candidatos: {len(candidates_by_genre)}")
print(f"Tamaño total aproximado: {total_mb:.2f} MB")

Videos candidatos: 10
Tamaño total aproximado: 31.96 MB


In [21]:
from urllib.request import urlopen
from shutil import copyfileobj

download_results = []

for candidate in candidates_by_genre.values():
    target_path = REFERENCE_DIR / candidate["file_name"]

    if target_path.exists():
        download_results.append({
            "file_name": candidate["file_name"],
            "status": "already_exists",
            "size_bytes": target_path.stat().st_size,
        })
        continue

    print(f"Descargando: {candidate['file_name']}")

    try:
        with urlopen(candidate["url"]) as response:
            with target_path.open("wb") as output_file:
                copyfileobj(response, output_file)

        download_results.append({
            "file_name": candidate["file_name"],
            "status": "downloaded",
            "size_bytes": target_path.stat().st_size,
        })

    except Exception as error:
        download_results.append({
            "file_name": candidate["file_name"],
            "status": "error",
            "error": str(error),
        })

download_results

Descargando: gBR_sBM_c01_d04_mBR0_ch01.mp4
Descargando: gHO_sBM_c01_d19_mHO0_ch01.mp4
Descargando: gJB_sBM_c01_d07_mJB0_ch01.mp4
Descargando: gJS_sBM_c01_d01_mJS0_ch01.mp4
Descargando: gKR_sBM_c01_d28_mKR0_ch01.mp4
Descargando: gLH_sBM_c01_d16_mLH0_ch01.mp4
Descargando: gLO_sBM_c01_d13_mLO0_ch01.mp4
Descargando: gMH_sBM_c01_d22_mMH0_ch01.mp4
Descargando: gPO_sBM_c01_d10_mPO0_ch01.mp4
Descargando: gWA_sBM_c01_d25_mWA0_ch01.mp4


[{'file_name': 'gBR_sBM_c01_d04_mBR0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3427560},
 {'file_name': 'gHO_sBM_c01_d19_mHO0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 2514732},
 {'file_name': 'gJB_sBM_c01_d07_mJB0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3560338},
 {'file_name': 'gJS_sBM_c01_d01_mJS0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3479034},
 {'file_name': 'gKR_sBM_c01_d28_mKR0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3416044},
 {'file_name': 'gLH_sBM_c01_d16_mLH0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3391897},
 {'file_name': 'gLO_sBM_c01_d13_mLO0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3392937},
 {'file_name': 'gMH_sBM_c01_d22_mMH0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3450461},
 {'file_name': 'gPO_sBM_c01_d10_mPO0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3434020},
 {'file_name': 'gWA_sBM_c01_d25_mWA0_ch01.mp4',
  'status': 'downloaded',
  'size_bytes': 3445005}]

In [22]:
downloaded_files = sorted(
    path for path in REFERENCE_DIR.iterdir()
    if path.is_file() and path.suffix.lower() == ".mp4"
)

print(f"Videos locales encontrados: {len(downloaded_files)}")

for path in downloaded_files:
    print(f"- {path.name}: {path.stat().st_size / (1024 ** 2):.2f} MB")

Videos locales encontrados: 10
- gBR_sBM_c01_d04_mBR0_ch01.mp4: 3.27 MB
- gHO_sBM_c01_d19_mHO0_ch01.mp4: 2.40 MB
- gJB_sBM_c01_d07_mJB0_ch01.mp4: 3.40 MB
- gJS_sBM_c01_d01_mJS0_ch01.mp4: 3.32 MB
- gKR_sBM_c01_d28_mKR0_ch01.mp4: 3.26 MB
- gLH_sBM_c01_d16_mLH0_ch01.mp4: 3.23 MB
- gLO_sBM_c01_d13_mLO0_ch01.mp4: 3.24 MB
- gMH_sBM_c01_d22_mMH0_ch01.mp4: 3.29 MB
- gPO_sBM_c01_d10_mPO0_ch01.mp4: 3.27 MB
- gWA_sBM_c01_d25_mWA0_ch01.mp4: 3.29 MB


In [23]:
import cv2

video_inventory = []

for video_path in downloaded_files:
    capture = cv2.VideoCapture(str(video_path))

    opened = capture.isOpened()
    frame_count = int(capture.get(cv2.CAP_PROP_FRAME_COUNT))
    fps = capture.get(cv2.CAP_PROP_FPS)
    width = int(capture.get(cv2.CAP_PROP_FRAME_WIDTH))
    height = int(capture.get(cv2.CAP_PROP_FRAME_HEIGHT))

    duration_sec = frame_count / fps if fps > 0 else None

    capture.release()

    video_inventory.append({
        "file_name": video_path.name,
        "opened": opened,
        "frame_count": frame_count,
        "fps": round(fps, 3),
        "width": width,
        "height": height,
        "duration_sec": round(duration_sec, 3) if duration_sec else None,
    })

for video in video_inventory:
    print(video)

{'file_name': 'gBR_sBM_c01_d04_mBR0_ch01.mp4', 'opened': True, 'frame_count': 719, 'fps': 59.94, 'width': 1920, 'height': 1080, 'duration_sec': 11.995}
{'file_name': 'gHO_sBM_c01_d19_mHO0_ch01.mp4', 'opened': True, 'frame_count': 523, 'fps': 59.94, 'width': 1920, 'height': 1080, 'duration_sec': 8.725}
{'file_name': 'gJB_sBM_c01_d07_mJB0_ch01.mp4', 'opened': True, 'frame_count': 719, 'fps': 59.94, 'width': 1920, 'height': 1080, 'duration_sec': 11.995}
{'file_name': 'gJS_sBM_c01_d01_mJS0_ch01.mp4', 'opened': True, 'frame_count': 719, 'fps': 59.94, 'width': 1920, 'height': 1080, 'duration_sec': 11.995}
{'file_name': 'gKR_sBM_c01_d28_mKR0_ch01.mp4', 'opened': True, 'frame_count': 719, 'fps': 59.94, 'width': 1920, 'height': 1080, 'duration_sec': 11.995}
{'file_name': 'gLH_sBM_c01_d16_mLH0_ch01.mp4', 'opened': True, 'frame_count': 719, 'fps': 59.94, 'width': 1920, 'height': 1080, 'duration_sec': 11.995}
{'file_name': 'gLO_sBM_c01_d13_mLO0_ch01.mp4', 'opened': True, 'frame_count': 719, 'fps':

## Fuente complementaria: AIST++

AIST++ se utilizará como fuente complementaria para revisar:

- Anotaciones 2D de pose.
- Confianza de los landmarks.
- Cobertura corporal.
- Relación entre secuencias y vistas de cámara.
- Referencia temporal para validar la extracción de pose.

AIST++ no será la fuente principal de videos RGB ni se utilizará para
entrenar un modelo de calidad de danza.

Las anotaciones 2D de AIST++ contienen 17 keypoints en formato COCO,
mientras que MediaPipe Pose proporciona 33 landmarks. Por lo tanto,
será necesario definir un mapeo entre ambos sistemas si se realiza una
comparación directa.

AIST++ utiliza anotaciones a 60 FPS exactos. Los videos RGB de AIST
presentan aproximadamente 59.94 FPS, por lo que esta diferencia deberá
considerarse durante la sincronización temporal.

In [24]:
aistpp_reference = {
    "source": "AIST++",
    "purpose": "Validación técnica de pose y cobertura corporal",
    "video_source": "AIST Dance Video Database",
    "keypoint_format": "COCO",
    "keypoint_count": 17,
    "annotation_fps": 60,
    "camera_views": 9,
    "used_for_quality_labels": False,
    "status": "Documentado; descarga de anotaciones pendiente",
}

aistpp_reference

{'source': 'AIST++',
 'purpose': 'Validación técnica de pose y cobertura corporal',
 'video_source': 'AIST Dance Video Database',
 'keypoint_format': 'COCO',
 'keypoint_count': 17,
 'annotation_fps': 60,
 'camera_views': 9,
 'used_for_quality_labels': False,
 'status': 'Documentado; descarga de anotaciones pendiente'}